shim.js

```diff
@@ -0,0 +1,89 @@
+// @info
+//   Polyfill for SVG 2 getPathData() and setPathData() methods. Based on:
+//   - SVGPathSeg polyfill by Philip Rogers (MIT License)
+//     https://github.com/progers/pathseg
+//   - SVGPathNormalizer by Tadahisa Motooka (MIT License)
+//     https://github.com/motooka/SVGPathNormalizer/tree/master/src
+//   - arcToCubicCurves() by Dmitry Baranovskiy (MIT License)
+//     https://github.com/DmitryBaranovskiy/raphael/blob/v2.1.1/raphael.core.js#L1837
+// @author
+//   Jarosław Foksa
+// @license
+//   MIT License
+
+const {
+	parsePathData, 
+	clonePathData,
+
+	getRectPathData,
+	getCirclePathData,
+	getEllipsePathData,
+	getLinePathData,
+	getPolylinePathData,
+	getPolygonPathData
+} = require('./polyfill');
+
+if (!SVGPathElement.prototype.getPathData || !SVGPathElement.prototype.setPathData) {
+	const setAttribute = SVGPathElement.prototype.setAttribute;
+  const removeAttribute = SVGPathElement.prototype.removeAttribute;
+
+  const $cachedPathData = Symbol();
+  const $cachedNormalizedPathData = Symbol();
+
+
+  SVGPathElement.prototype.setAttribute = function(name, value) {
+    if (name === "d") {
+      this[$cachedPathData] = null;
+      this[$cachedNormalizedPathData] = null;
+    }
+
+    setAttribute.call(this, name, value);
+  };
+
+  SVGPathElement.prototype.removeAttribute = function(name, value) {
+    if (name === "d") {
+      this[$cachedPathData] = null;
+      this[$cachedNormalizedPathData] = null;
+    }
+
+    removeAttribute.call(this, name);
+  };
+
+  SVGPathElement.prototype.getPathData = function(options) {
+    if (options && options.normalize) {
+      if (!this[$cachedNormalizedPathData]) {
```

Comment:
I noticed how you double-cached the normalized path, but I don't think it's worth